### PR TITLE
Let fetch_tree also returns position and parent_guid

### LINKED ISSUE #808: fetch_tree should return position and parentGUID information
As it is `fetch_public_tree` has to fix it up afterwards, which both takes additional query and is a pain.

Worth keeping in mind https://github.com/mozilla/application-services/issues/807 also, since the main reason this is difficult is that it does not return a type that can store parentGUID / position.

Follow up from #743.

### PR DESCRIPTION
**Pull request by lyuyuan92**
fixes #808

### CODE REVIEW

**Comment by lyuyuan92** [comment]
Sure, I will add tests for `fetch_tree` function. If somewhere else could benefit from adding the 2 fields to `BookmarkTreeNode`, please let me know. I am willing to make the change.

**Comment by thomcc** [comment]
> If somewhere else could benefit from adding the 2 fields to `BookmarkTreeNode`, please let me know. I am willing to make the change.

No, for now this is fine (this function isn't used elsewhere), and is a lot simpler and easy for me to review than had you modified the big fetch_tree query.

File: components/places/src/storage/bookmarks.rs [hunk 1]
```diff
@@ -1124,7 +1124,10 @@ fn inflate(
 
 /// Fetch the tree starting at the specified folder guid.
 /// Returns a BookmarkTreeNode::Folder(_)
```
Comment (thomcc) [inline components/places/src/storage/bookmarks.rs]: Ah, one more thing, this should be updated.
```suggestion
/// Returns a `BookmarkTreeNode::Folder(_)`, its parent's guid (if any), and
/// position inside its parent.
```

Comment (lyuyuan92) [reply components/places/src/storage/bookmarks.rs]: Please re-review whenever you get a chance, thanks!

**Pull request merged**